bin/runtime/src/lib.rs

```diff
@@ -825,18 +826,18 @@ impl InstanceFilter<RuntimeCall> for ProxyType {
                         | RuntimeCall::NominationPools(..)
                 )
             }
+            ProxyType::Nomination => {
+                matches!(
+                    c,
+                    RuntimeCall::NominationPools(pallet_nomination_pools::Call::nominate { .. })
+                        | RuntimeCall::Staking(pallet_staking::Call::nominate { .. })
+                )
+            }
         }
     }
     fn is_superset(&self, o: &Self) -> bool {
-        // ProxyType::Staking ⊆ ProxyType::NonTransfer ⊆ ProxyType::Any
-        match (self, o) {
-            (ProxyType::Any, _) => true,
-            (_, ProxyType::Any) => false,
-            (ProxyType::NonTransfer, ProxyType::Staking) => true,
-            (ProxyType::Staking, ProxyType::NonTransfer) => false,
-            (ProxyType::Staking, ProxyType::Staking) => true,
-            (ProxyType::NonTransfer, ProxyType::NonTransfer) => true,
-        }
+        // ProxyType::Nomination ⊆ ProxyType::Staking ⊆ ProxyType::NonTransfer ⊆ ProxyType::Any
+        (*self as u64) <= (*o as u64)
```

Comment:
The reason why this was verbose is to make it safer to modify this code -- i.e., add new proxy types. What we want to avoid is someone adding a new proxy type and then the compiler not telling them `is_superset` requires attention.

I don't know what's the best way to make this right though...

Reply:
Btw. this piece of code is 100% security critical. If we get this wrong, it's really bad :P

Reply:
Oh, right. I forgot about this when reviewing. I would suggest macro to generate arms like `is_superset!(Any < ... < Nomination)` but I dont like macros :P

Reply:
Added exhaustive matching (and improved readability)